components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserMgtAuditLogger.java

```diff
@@ -18,144 +18,351 @@
         return true;
     }
 
-    public boolean doPostAddRole(String roleName, String[] userList, Permission[] permissions, UserStoreManager
-            userStoreManager) throws UserStoreException {
+    @Override
+    public boolean doPreDeleteUserClaimValues(String userName, String[] claims, String profileName,
+            UserStoreManager userStoreManager) {
 
-        if(!isEnable()) {
+        if (!isEnable()) {
             return true;
         }
 
-        audit.info(String.format(AUDIT_MESSAGE, getUser(), "Add Role", roleName, "Users : "
-                + Arrays.toString(userList) + " Permissions : " + Arrays.toString(permissions), SUCCESS));
+        JSONObject dataObject = new JSONObject();
+        dataObject.put(ListenerUtils.PROFILE_FIELD, profileName);
+        dataObject.put(ListenerUtils.CLAIMS_FIELD, new JSONObject(claims));
+        audit.warn(
+                createAuditMessage(ListenerUtils.DELETE_USER_CLAIM_VALUES_ACTION, userName, dataObject, IN_PROGRESS));
         return true;
     }
 
-    public boolean doPostUpdateRoleName(String roleName, String newRoleName, UserStoreManager userStoreManager)
-            throws UserStoreException {
+    @Override
+    public boolean doPostDeleteUserClaimValues(String userName, UserStoreManager userStoreManager) {
 
-        if(!isEnable()) {
+        if (!isEnable()) {
             return true;
         }
 
-        audit.info(String.format(AUDIT_MESSAGE, getUser(), "Update Role Name", roleName,
-                "Old : " + roleName + " New : " + newRoleName, SUCCESS));
+        audit.warn(createAuditMessage(ListenerUtils.DELETE_USER_CLAIM_VALUES_ACTION, userName, null, SUCCESS));
         return true;
     }
 
+    @Override
+    public boolean doPreDeleteUserClaimValue(String userName, String claimURI, String profileName,
+            UserStoreManager userStoreManager) {
+
+        if (!isEnable()) {
+            return true;
+        }
+
+        JSONObject dataObject = new JSONObject();
+        dataObject.put(ListenerUtils.CLAIM_URI_FIELD, claimURI);
+        dataObject.put(ListenerUtils.PROFILE_FIELD, profileName);
+        audit.warn(createAuditMessage(ListenerUtils.DELETE_USER_CLAIM_VALUE_ACTION, userName, dataObject, IN_PROGRESS));
+        return true;
+    }
+
+    @Override
+    public boolean doPostDeleteUserClaimValue(String userName, UserStoreManager userStoreManager) {
+
+        if (!isEnable()) {
+            return true;
+        }
+
+        audit.warn(createAuditMessage(ListenerUtils.DELETE_USER_CLAIM_VALUE_ACTION, userName, null, SUCCESS));
+        return true;
+    }
+
+    @Override
+    public boolean doPostUpdateCredential(String userName, Object credential, UserStoreManager userStoreManager) {
+
+        if (!isEnable()) {
+            return true;
+        }
+
+        audit.warn(createAuditMessage(ListenerUtils.CHANGE_PASSWORD_BY_USER_ACTION, userName, null, SUCCESS));
+        return true;
```

Comment:
Can't we make the code as follows:- 
if (isEnable()) {
   audit.warn(createAuditMessage(ListenerUtils.CHANGE_PASSWORD_BY_USER_ACTION, userName, null, SUCCESS));
}
return true;

So that we can have only one return statement.